### PR TITLE
Add Microsoft Research Sequential Question Answering (SQA) Dataset

### PR DESCRIPTION
**Pull request by mattbui**
For more information: https://msropendata.com/datasets/b25190ed-0f59-47b1-9211-5962858142c2

### CODE REVIEW

**Comment by thomwolf** [comment]
I proposed something a few weeks ago in #898 (un-merged) but I think that the way that @mattbui added the dataset in the present PR is smarter and simpler should replace my PR #898.

(Narrator voice: *And it was around that time that Thomas realized that the community was now a lot smarter than him and he should hand-over the library he had started with @lhoestq to the community and stop pretending he knew everything about it.*)

File: datasets/msr_sqa/msr_sqa.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            header = lines[0].strip().split(",")
+            data = [line.strip().split(",") for line in lines[1:]]
+            return header, data
+
+        with open(filepath, encoding="utf-8") as f:
+            reader = csv.DictReader(f, delimiter="\t")
+            for row in reader:
+                item = dict(row)
+                item["answer_text"] = eval(item["answer_text"])
+                item["answer_coordinates"] = eval(item["answer_coordinates"])
```
Comment (lhoestq) [inline datasets/msr_sqa/msr_sqa.py:109]: Could you parse their values without using `eval` ?
`eval` is considered unsafe

**Review by lhoestq — COMMENTED** [review]
That's really cool thanks !

I just left one comment about the use of `eval`

File: datasets/msr_sqa/msr_sqa.py (hunk 1 above)
Comment (NielsRogge) [reply datasets/msr_sqa/msr_sqa.py:109]: Perhaps you can use the methods I proposed [here](https://github.com/huggingface/datasets/issues/880#issuecomment-732939750), which are used in the original implementation of TAPAS by Google AI (which achieves SOTA on SQA).

Comment (mattbui) [reply datasets/msr_sqa/msr_sqa.py]: @NielsRogge Thanks for the suggestion 💡

**Review by lhoestq — APPROVED** [review]
Awesome thank you :) 

Thanks for suggesting `ast.literal_eval` @NielsRogge 
It's all clean now ! thank you @mattbui

**Pull request merged**